### PR TITLE
Update ci workflow

### PR DESCRIPTION
**Pull request by gnokoheat**
### separate main branch from ci workflow
- (push) main branch : docker build & push (docker hub)
- (push) not main branch : lint, unit test, docker build & push (google cloud)
### use docker default tag (latest)
```
# docker pull taehoio/youtube2notion
Using default tag: latest
Error response from daemon: manifest for taehoio/youtube2notion:latest not found: manifest unknown: manifest unknown
```

Related https://github.com/taehoio/youtube2notion/issues/14

### CODE REVIEW

File: .github/workflows/ci-main.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        id: docker_build
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: |
+            taehoio/youtube2notion:latest
```
Comment (xissy) [inline .github/workflows/ci-main.yml]:
```suggestion
          tags: |
            taehoio/youtube2notion:latest
            taehoio/youtube2notion:${{ github.sha }}
            
```
not to miss previous container images.

**Review by xissy — APPROVED** [review]
LGTM. please address my nit request to have `github.sha` as an image tag.

**Comment by xissy** [comment]
just to give a little context why I didn't put the `latest` tag for the container image, I like to specify all kinds of service versions or image tags explicitly. sometimes (often actually) `latest` tag name occurs image version conflicts which are very hard to track and debug. but in this case, providing `latest` tag seems to be valid since it's open-sourced and has to support common users' needs.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,7 +1,10 @@
 name: ci
 
 on:
-  - push
+  push:
+    branches:
+      - '*'
+      - '!main'
```
Comment (xissy) [inline .github/workflows/ci.yml]:
```suggestion
```

**Review by xissy — CHANGES_REQUESTED** [review]
hmm... one nit thing here. 

I'd like to not skip `lint` and `test` when a pr is merged into `main` branch. Since we've used `squash and merge` policy for this repo, it would be better to have those to make sure there is no issue when a pr is merged.

**Review by xissy — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (gnokoheat) [reply .github/workflows/ci.yml]: 확인 감사합니다! 관련해서 시간 되실 때 레포지토리 `Branch protection rule` 에서 `main` 브랜치의 `Require pull request reviews before merging` 설정도 부탁 드립니다 🙏

**Pull request merged**